lib/spack/spack/cmd/create.py

```diff
@@ -98,109 +97,160 @@ class ${class_name}(Package):
 
 ${dependencies}
 
-    def install(self, spec, prefix):
-${install}
+${body}
 """)
 
-# Build dependencies and extensions
-dependencies_dict = {
-    'autotools': """\
-    # FIXME: Add dependencies if required.
-    # depends_on('foo')""",
 
-    'cmake': """\
-    # FIXME: Add additional dependencies if required.
-    depends_on('cmake', type='build')""",
+class DefaultGuess(object):
 
-    'scons': """\
-    # FIXME: Add additional dependencies if required.
-    depends_on('scons', type='build')""",
+    base_class_name = 'Package'
 
-    'bazel': """\
-    # FIXME: Add additional dependencies if required.
-    depends_on('bazel', type='build')""",
+    dependencies = """\
+    # FIXME: Add dependencies if required.
+    # depends_on('foo')"""
 
-    'python': """\
-    extends('python')
+    body = """\
+    def install(self, spec, prefix):
+        # FIXME: Unknown build system
+        make()
+        make('install')"""
 
-    # FIXME: Add additional dependencies if required.
-    # depends_on('py-setuptools', type='build')
-    # depends_on('py-foo',        type=nolink)""",
+    def __init__(self, name, url, version_hash_tuples):
+        self.name = name
+        self.class_name = mod_to_class(name)
+        self.url = url
+        self.version_hash_tuples = version_hash_tuples
 
-    'R': """\
-    extends('R')
+    @property
+    def versions(self):
+        """Adds a version() call to the package for each version found."""
+        max_len = max(len(str(v)) for v, h in self.version_hash_tuples)
+        format = "    version(%%-%ds, '%%s')" % (max_len + 2)
+        return '\n'.join(
+            format % ("'%s'" % v, h) for v, h in self.version_hash_tuples
+        )
 
-    # FIXME: Add additional dependencies if required.
-    # depends_on('r-foo', type=nolink)""",
 
-    'octave': """\
-    extends('octave')
+class AutotoolsGuess(DefaultGuess):
 
-    # FIXME: Add additional dependencies if required.
-    # depends_on('octave-foo', type=nolink)""",
+    base_class_name = 'AutotoolsPackage'
 
-    'unknown': """\
+    dependencies = """\
     # FIXME: Add dependencies if required.
+    # depends_on('m4', type='build')
+    # depends_on('autoconf', type='build')
+    # depends_on('automake', type='build')
+    # depends_on('libtool', type='build')
     # depends_on('foo')"""
-}
 
-# Default installation instructions
-install_dict = {
-    'autotools': """\
-        # FIXME: Modify the configure line to suit your build system here.
-        configure('--prefix={0}'.format(prefix))
+    body = """\
+    def configure_args(self):
+       # FIXME : Add arguments other than --prefix
+       # FIXME : If not needed delete the function
```

Comment:
@adamjstewart I'll keep that in mind when writing TODOs or FIXMEs here. 

> There aren't generally spaces before colons in English, although I think it's different for French

Eheh, it's just a bad habit I kept in a project I worked before (lucky me I am Italian, French is way too difficult to learn properly!  :smile: )